File: probes/bashlex/probe.py
# Probe: can bashlex parse each line of the shared corpus?
import sys
import bashlex

for line in open(sys.argv[1]).read().strip().split("\n"):
    name, cmd = line.split("\t")
    try:
        bashlex.parse(cmd)
        print(f"OK    {name}")
    except Exception as e:
        print(f"FAIL  {name:<24} {type(e).__name__}: {str(e).splitlines()[0]}")

File: probes/bash-parser/probe.js
// Probe: can bash-parser (npm) parse each line of the shared corpus?
const parse = require('bash-parser');
const fs = require('fs');

for (const line of fs.readFileSync(process.argv[2], 'utf8').trim().split('\n')) {
  const [name, cmd] = line.split('\t');
  try {
    parse(cmd, { insertLOC: true });
    console.log(`OK    ${name}`);
  } catch (e) {
    console.log(`FAIL  ${name.padEnd(24)} ${String(e.message).split('\n')[0]}`);
  }
}
